crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2039,6 +2025,16 @@ impl<'src> Parser<'src> {
     fn parse_type_param(&mut self) -> ast::TypeParam {
         let start = self.node_start();
 
+        // TODO(dhruvmanila): CPython throws an error if `TypeVarTuple` or `ParamSpec`
+        // has bounds:
+        //
+        //    type X[*T: int] = int
+        //             ^^^^^
+        // SyntaxError: cannot use bound with TypeVarTuple
+        //
+        // We should do the same but currently we can't without throwing away the parsed
+        // expression because the AST can't contain it.
```

Comment:
What does "the AST can't contain it" mean here?

Reply:
There's no field to store an expression for `TypeVarTuple`:

https://github.com/astral-sh/ruff/blob/a516f8f1d86cfb49bc3a395cbe8a3af02857adc5/crates/ruff_python_ast/src/nodes.rs#L3156-L3161

while for `TypeVar` there is:

https://github.com/astral-sh/ruff/blob/a516f8f1d86cfb49bc3a395cbe8a3af02857adc5/crates/ruff_python_ast/src/nodes.rs#L3130-L3135

We'd need to parse the expression to highlight the correct range. Or, we could just check if there's a `:` token and provide the error message highlighting the `:` token.

Reply:
I don't think I understand the comment. Does it mean we don't emit a syntax error or is it that Python emits a more specific syntax error?

Reply:
The latter. We do emit a syntax error. This comment is for a better error message if we want.